magento/magento2
# magento/magento2#24730: Fix Frontend Invoice PDF configured Logo image

### Description (*)
According to https://github.com/magento/magento2/issues/24730 Admin Configuration Stores -> Configuration -> Sales -> Sales ->'Invoice and Packaging Slip Design' -> 'Logo for HTML Print View' does not affect on logo on frontend invoice print page.

sales_order_printinvoice page use general theme class/template for Logo generation - app/code/Magento/Theme/Block/Html/Header/Logo.php

In Block 'logo_src' argument is checked, before logo path is generated, so I use it as extension point to insert logo configuration on print invoice page. 
Changes affects only frontend sales order print invoice page through layout file. Similar approach can be used to change other print pages, like sales_order_print.
 

### Related Pull Requests
no

### Fixed Issues (if relevant)

1. Fixes magento/magento2#24730

### Manual testing scenarios (*)

1. Create Order from storefront.
2. Create Invoice from admin.
3. On storefront go to Customer Account > My Orders > View Order
4. Select Invoices Tab.
5. Select Print Invoice. By Default Logo is Luma.
6. In Admin Configuration go to Sales -> Sales ->'Invoice and Packaging Slip Design' -> 'Logo for HTML Print View' and upload custom image
7. Clear cache.
8. In customer account repeat steps 3-5. The Logo now should be your custom image.

### Questions or comments

no

### Contribution checklist (*)
